### DISCUSSION TITLE
Runtime error from Cross Compile of Python

### DISCUSSION THREAD

**Discussion by GRRedWings** [android/ndk#1965] [Q&A]
I have been maintaining a project that builds the latest Python to run on Android.  With the latest NDK and Python 3.12, I now get an error when trying to run Python

dlopen failed: cannot locate symbol "PyExc_OSError" referenced by "/data/data/Python64/lib-dynload/_socket.cpython-312.so"

In looking at other errors that say they cannot locate symbol there is a lot of discussion on the API version. I have tried building this with API 23, and 26. I believe that the latest NDK has made using API23 more difficult, but I can build against API 26 without much difficulty.

Here is the project that I have been working on, and pointed at the latest branch.
https://github.com/GRRedWings/python3-android/tree/DWE-3790

 I have looked at socketmodule where I believe the _socket.cpython-3.12.so is being built from. I see references to PyExc_OSError which is defined in pyerrors.h as PyAPI_DATA(PyObject *) PyExc_OSError;
 
 Thanks in advance for any help or insight.

**Comment by mhsmith** [comment]
See https://github.com/python/cpython/issues/111225.
